src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -1,22 +1,22 @@
 package frc.robot.subsystems;  
 
 import com.ctre.phoenix.motorcontrol.ControlMode;
-import com.ctre.phoenix.motorcontrol.can.TalonSRX;
-
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.*;
 
 public class Drivetrain {
 
-    private static TalonSRX leftFront;
-    private static TalonSRX leftBack;
-    private static TalonSRX rightFront;
```

Comment:
Now it's right, right?

Reply:
Why change them to WPI_TalonSRX? There's no need really. The TalonSRX libraries work just fine. I don't understand why the drive controls need to be threaded. Just by the way, a thread.sleep in the driver controls means that for the time it is sleeping the robot is not under control of the program because the program is asleep. That's why they disable your robot, it's considered to be going haywire. And trust me, those 20ms matter. The code is on a 20ms loop and threaded automatically, so why not just allow the background code to do what it was written to do? Threads are only okay in robot code when you are doing calculations ont he rio and you don't want them to take up too much bandwidth, but usually you just use a seperate processor (ie a raspberry pi) to prevent that issue because it is still a dangerous game to play.

Reply:
The whole program is not asleep. The master thread is not sleeping, only the thread running driver controls.

I don't see documentation on how its threaded automatically. How is it done? The 20ms loop is so that the loop constantly checking the controller doesn't take up too much network bandwidth.

Reply:
Only the thread that controls the drivetrain, which is dangerous. It is automatically threaded at 20ms so you don't need to have a thread within that thread to make it the same time structure. Just don't thread it. 

I don't know how it's done, I haven't delved deep enough into the background code to know how exactly it is put together, but I have broached the topic before with FTA's and mentors who have been around since the start of modern FMS and I'm telling you that's how it works. Don't thread robot code unless you have some very complex calculations going on, the RobotRio can handle it.